### PR TITLE
Resource Management Control

### PR DESCRIPTION
**Pull request by toniyanchev**

### CODE REVIEW

File: microservices/resource-management-control/src/service/index.js [hunk 1]
```diff
@@ -1,9 +1,37 @@
 const service = require('./service');
+const grpc = require('grpc');
+const errorHandler = require('../errorHandler.js');
 
 exports.createLecture = async (call, callback) => {
+    const { courseId, name } = call.request;
+    if (!courseId || courseId == null) {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/index.js:7]: only !courseId is enough

File: microservices/resource-management-control/src/service/index.js [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+        const status = grpc.status.INVALID_ARGUMENT;
+        return errorHandler(callback, status, 'Blank request');
+    }
+    if (!name || name == null) {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/index.js:11]: here too

File: microservices/resource-management-control/src/service/index.js [hunk 3]
```diff
@@ -1,9 +1,37 @@
...
+        const status = grpc.status.INVALID_ARGUMENT;
+        return errorHandler(callback, status, 'Blank request');
+    }
+
     service.createLecture(call, callback);
 };
 
+exports.getLecture = async (call, callback) => {
+    const { id } = call.request;
+    if (!id || id == null) {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/index.js:21]: here too

File: microservices/resource-management-control/src/service/index.js [hunk 4]
```diff
@@ -1,9 +1,37 @@
 const service = require('./service');
+const grpc = require('grpc');
+const errorHandler = require('../errorHandler.js');
 
 exports.createLecture = async (call, callback) => {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/index.js:5]: this functions are not async. Remove it

File: microservices/resource-management-control/src/service/index.js [hunk 5]
```diff
@@ -1,9 +1,37 @@
...
+        const status = grpc.status.INVALID_ARGUMENT;
+        return errorHandler(callback, status, 'Blank request');
+    }
+
+    service.getLecture(call, callback);
+};
+
 exports.getAllLectures = async (call, callback) => {
+    const { courseId } = call.request;
+    if (!courseId || courseId == null) {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/index.js:31]: fix

File: microservices/resource-management-control/src/service/index.js [hunk 6]
```diff
@@ -1,9 +1,37 @@
...
+    const { id } = call.request;
+    if (!id || id == null) {
+        const status = grpc.status.INVALID_ARGUMENT;
+        return errorHandler(callback, status, 'Blank request');
+    }
+
+    service.getLecture(call, callback);
+};
+
 exports.getAllLectures = async (call, callback) => {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/index.js:29]: fix

File: microservices/resource-management-control/src/service/index.js [hunk 7]
```diff
@@ -1,9 +1,37 @@
...
+    }
+    if (!name || name == null) {
+        const status = grpc.status.INVALID_ARGUMENT;
+        return errorHandler(callback, status, 'Blank request');
+    }
+
     service.createLecture(call, callback);
 };
 
+exports.getLecture = async (call, callback) => {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/index.js:19]: fix

File: microservices/resource-management-control/src/service/service.js [hunk 8]
```diff
@@ -1,54 +1,101 @@
...
+const grpc = require('grpc');
 
 const { getRepository } = require('typeorm');
 const { getConnection } = require('typeorm');
 
-// const errorHandler = require('../util/errorHandler');
+const errorHandler = require('../errorHandler');
 const Lecture = require('../entity/LectureSchema.js');
 const Resource = require('../entity/ResourceSchema.js');
+/* eslint-disable-next-line no-unused-vars */
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/service.js:9]: why have you added eslint-disable?

File: microservices/resource-management-control/src/service/service.js [hunk 9]
```diff
@@ -1,54 +1,101 @@
...
+        const status = grpc.status.INTERNAL;
+        return errorHandler(callback, status, 'Database error', error);
+    }
+    if (lecture == null) {
+        const status = grpc.status.INVALID_ARGUMENT;
+        return errorHandler(callback, status, 'No lecture with this id');
+    }
+
+    const token = jwt.sign(
+        { resourceId: lecture.resourceId },
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/service.js:73]: I hope you have fixed column name to be resource_id

File: microservices/resource-management-control/src/service/service.js (hunk 8 above)
Comment (toniyanchev) [reply microservices/resource-management-control/src/service/service.js:9]: cuz it says i dont use jwt

File: microservices/resource-management-control/src/service/service.js [hunk 10]
```diff
@@ -1,54 +1,101 @@
...
+    const lectureRepository = getRepository(Lecture);
 
+    let lecture;
+    try {
+        lecture = await lectureRepository.findOne(id);
+    } catch (error) {
+        const status = grpc.status.INTERNAL;
+        return errorHandler(callback, status, 'Database error', error);
+    }
+    if (lecture == null) {
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/service.js:67]: never compare with null or 0 except you insist on boolean type
write just if (lecture)

File: microservices/resource-management-control/src/service/service.js [hunk 11]
```diff
@@ -1,54 +1,101 @@
...
+    }
+
+    const token = jwt.sign(
+        { resourceId: lecture.resourceId },
+        process.env.JWT_SECRET,
+        {
+            expiresIn: process.env.JWT_VALID_TIME
+        }
+    );
+    const url = process.env.RESOURCE_URL + '/download/' + token;
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/service.js:79]: env var is called RESOURCE_MANAGEMENT_URL

File: microservices/resource-management-control/src/service/service.js (hunk 9 above)
Comment (toniyanchev) [reply microservices/resource-management-control/src/service/service.js]: i Did

File: microservices/resource-management-control/src/service/service.js (hunk 10 above)
Comment (toniyanchev) [reply microservices/resource-management-control/src/service/service.js:67]: think u meant !lecture

Comment (toniyanchev) [reply microservices/resource-management-control/src/service/service.js:67]: and fixed it

File: microservices/resource-management-control/src/service/service.js [hunk 12]
```diff
@@ -1,54 +1,105 @@
-// const grpc = require('grpc');
+const grpc = require('grpc');
 
 const { getRepository } = require('typeorm');
```
Comment (samigeorgiev) [inline microservices/resource-management-control/src/service/service.js:3]: combine this imports const { getRepository, getConnection } = require('typeorm');

**Pull request merged**